app/questionnaire/questionnaire_schema.py

```diff
@@ -0,0 +1,168 @@
+from app.validation.error_messages import error_messages
+
+
+class QuestionnaireSchema(object):
+
+    def __init__(self, questionnaire_json):
+        self.json = questionnaire_json
+        self._groups = self._get_groups()
+        self._blocks = self._get_blocks()
```

Comment:
I think it would be clearer if we called the properties `_groups_by_id` and `blocks_by_id` and renamed the methods `_get_groups_by_id` and `_get_blocks_by_id` to make a distinction between these methods and the `get_groups` and `get_blocks` methods